src/Core/AcmeClient.php

```diff
@@ -328,6 +332,7 @@ protected function requestResource(string $method, string $resource, array $payl
     private function createCertificateResponse(CertificateRequest $csr, string $certificate): CertificateResponse
     {
         $certificatesChain = null;
+        $certificate = str_replace("-----END CERTIFICATE-----\n-----BEGIN CERTIFICATE-----", "-----END CERTIFICATE-----\n\n-----BEGIN CERTIFICATE-----", $certificate);
```

Comment:
What's the aim of that?

Reply:
True, a little more explanation is needed - the aim is to make `getCertificate()->getPem()` & `getCertificate()->getIssuerCertificate()->getPEM()` working as expected.

Let's Encrypt returns the chain certs separated by a new line. That is why your code `explode("\n\n", $certificate)` works. But regards RFC there is no such a separation, casing the above function to failed when ZeroSSL or BuyPASS CAs are used.

One single (but yes, dirty) line of code makes the functions mentioned working as expected for all three CAs. Otherwise more sophisticated parser is needed to split leave certificate from all the rest.